### PR TITLE
Added Junit tests for AddForexParser and updated error message for invalid fx name

### LINKED ISSUE #34: Add forex tests
Add tests to check input parameters for adding forex

### PR DESCRIPTION
**Pull request by KVignesh122**
Fix #34

### CODE REVIEW

File: src/main/java/seedu/mtracker/console/AddInstrumentParser.java [hunk 1]
```diff
@@ -33,11 +34,19 @@ public static String getInstrumentNameFromUser(String instrumentType) {
...
+            } else if (name.isEmpty()) {
                 throw new IllegalArgumentException();
             }
         } catch (IllegalArgumentException e) {
-            ErrorMessage.displayAddInstrumentNameError(instrumentType);
+            if (instrumentType.equals(AddForexParser.INSTRUMENT_TYPE)) {
+                ErrorMessage.displayAddForexNameError();
+            } else {
+                ErrorMessage.displayAddInstrumentNameError(instrumentType);
+            }
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/console/AddInstrumentParser.java:49]: Can do this in a separate PR, the pr was suppose to be for junit specific changes.
On a separate note, should we do this name validation? Cause like stock we don't rly care if user puts "AAPL" vs "apple" cause we cant check with a known db of valid instrument names. For forex, even if we force user to put 6 characters they can still put this rubbish like aaaaaa. So I was thinking maybe we don't do this check and we just put a note on user guide that its up to the user responsibility to put in accurate info

File: src/test/java/seedu/mtracker/console/AddForexParserTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import org.junit.jupiter.api.Test;
+
+import java.io.ByteArrayInputStream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class AddForexParserTest {
+
+    public static final int PARAMETER_SIZE = 7;
+    private static final String STR_SEP = "%1$s";
```
Comment (theodorekwok) [inline src/test/java/seedu/mtracker/console/AddForexParserTest.java:12]: Rename to SEPARATOR_SPECIFIERS or something %1$s is like a format specifier

File: src/test/java/seedu/mtracker/console/AddForexParserTest.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    public static final String USER_INPUT_TRY_INVALID_PRICE = STR_SEP + "TTTXXX"
+            + STR_SEP + "lol"
+            + STR_SEP + "0.81"
+            + STR_SEP + "negative"
+            + STR_SEP + "0.79"
+            + STR_SEP + "0.70"
+            + STR_SEP + "20 Oct"
+            + STR_SEP + "fooRemarks";
+
+    public static final String USER_INPUT_TRY_INVALID_SENTIMENT = STR_SEP + "TTTXXX"
```
Comment (theodorekwok) [inline src/test/java/seedu/mtracker/console/AddForexParserTest.java:68]: The junit tests currently doesnt test for invalid entry price and exit price given that you said those parameters are compulsory. Create a new issue for this and don't forget to include the validations for those params

**Review by theodorekwok — COMMENTED** [review]
Good job, only have naming issue and some notes about the junit tests. Once resolved can merge alr

File: src/test/java/seedu/mtracker/console/AddForexParserTest.java (hunk 3 above)
Comment (KVignesh122) [reply src/test/java/seedu/mtracker/console/AddForexParserTest.java:68]: Ok

File: src/test/java/seedu/mtracker/console/AddForexParserTest.java (hunk 2 above)
Comment (KVignesh122) [reply src/test/java/seedu/mtracker/console/AddForexParserTest.java:12]: Done

File: src/main/java/seedu/mtracker/console/AddInstrumentParser.java (hunk 1 above)
Comment (theodorekwok) [reply src/main/java/seedu/mtracker/console/AddInstrumentParser.java:50]: I still think the length check is redundant, like I write "usd to sgd" its like somewhat understandable to user. The point is like if not possible to validate a name completely I feel its pointless to add the check

File: src/main/java/seedu/mtracker/console/AddInstrumentParser.java [hunk 4]
```diff
@@ -34,11 +35,19 @@ public static String getInstrumentNameFromUser(String instrumentType) {
     public static boolean isValidName(String name, String instrumentType) {
         boolean isValid = true;
         try {
-            if (name.isEmpty()) {
+            if (instrumentType.equals(AddForexParser.INSTRUMENT_TYPE)) {
+                if (name.length() != FX_PAIR_NAME_LENGTH) {
+                    throw new IllegalArgumentException();
+                }
+            } else if (name.isEmpty()) {
```
Comment (theodorekwok) [inline src/main/java/seedu/mtracker/console/AddInstrumentParser.java:42]:
```suggestion
            if (name.isEmpty() || (instrumentType.equals(AddForexParser.INSTRUMENT_TYPE) && name.length() != FX_PAIR_NAME_LENGTH)) {
               throw new IllegalArgumentException();
```

Simplify the conditional, to make the code better maybe abstract out the conditional here

File: src/main/java/seedu/mtracker/console/AddInstrumentParser.java (hunk 1 above)
Comment (KVignesh122) [reply src/main/java/seedu/mtracker/console/AddInstrumentParser.java:50]: Nah man forex names are not <smthin> to <smthin>. Its always like USDSGD, AUDUSD, EURJPY, etc. so I want to keep this length check.

**Review by theodorekwok — APPROVED** [review]
LGTM

**Pull request merged**